typewriter/fixes/tests/base_py2.py

```diff
@@ -227,6 +227,102 @@ class MyClass: pass
 
+    def test_type_by_mod_import_as(self):
+        self.setTestData(
+            [{"func_name": "nop",
+              "path": "mod1.py",
+              "line": 1,
+              "signature": {
+                  "arg_types": ["mod1.MyClass"],
+                  "return_type": "mod2.AnotherClass"},
+              }])
+        a = """\
+            import mod2 as bar
+            def nop(foo):
+                return bar.AnotherClass()
+            class MyClass: pass
+            """
+        b = """\
+            import mod2 as bar
+            def nop(foo):
+                # type: (MyClass) -> bar.AnotherClass
+                return bar.AnotherClass()
+            class MyClass: pass
+            """
+        self.check(a, b)
+
+    def test_type_by_dotted_import_as(self):
+        self.setTestData(
+            [{"func_name": "nop",
+              "path": "mod1.py",
+              "line": 1,
+              "signature": {
+                  "arg_types": ["mod1.MyClass"],
+                  "return_type": "pkg.mod2.AnotherClass"},
+              }])
+        a = """\
+            import pkg.mod2.AnotherClass as bar
+            def nop(foo):
+                return bar()
+            class MyClass: pass
+            """
+        b = """\
+            import pkg.mod2.AnotherClass as bar
+            def nop(foo):
+                # type: (MyClass) -> bar
+                return bar()
+            class MyClass: pass
+            """
+        self.check(a, b)
+
+    def test_type_by_import_as(self):
+        self.setTestData(
+            [{"func_name": "nop",
+              "path": "mod1.py",
+              "line": 1,
+              "signature": {
+                  "arg_types": ["mod1.MyClass"],
+                  "return_type": "mod2.AnotherClass"},
+              }])
+        a = """\
+            from mod2 import AnotherClass as bar
+            def nop(foo):
+                return bar()
+            class MyClass: pass
+            """
+        b = """\
+            from mod2 import AnotherClass as bar
+            def nop(foo):
+                # type: (MyClass) -> bar
+                return bar()
+            class MyClass: pass
+            """
+        self.check(a, b)
+
+    def test_type_by_from_import_as_with_dotted_package(self):
+        self.setTestData(
+            [{"func_name": "nop",
+              "path": "mod1.py",
+              "line": 1,
+              "signature": {
+                  "arg_types": ["mod1.MyClass"],
+                  "return_type": "pkg.mod2.AnotherClass"},
+              }])
+        a = """\
+            from pkg.mod2 import AnotherClass as bar
+            def nop(foo):
+                return bar()
+            class MyClass: pass
+            """
+        b = """\
+            from pkg.mod2 import AnotherClass as bar
+            def nop(foo):
+                # type: (MyClass) -> bar
+                return bar()
+            class MyClass: pass
+            """
+        self.check(a, b)
+
```

Comment:
Since we've eliminated the kwarg to control whether we consider aliases a match, I want to make sure that we're getting the right result, when there could be potential confusion between aliases and import names.

I'd like to add a test like this to make sure adding these extra import lines doesn't change the results:

```
# Attempt to disguise as bar - before real import
import bar as tofu
import mod3.bar
import mod3.bar as thing
from mod3 import bar as other
from mod3.submod import bar as blah
from mod3.bar import bar as tricksy

# real import
from mod2 import AnotherClass as bar

# Attempt to disguise as bar - after real import
import bar as tofu2
import mod4.bar
import mod4.bar as thing2
from mod4 import bar as other2
from mod4.submod import bar as blah2
from mod4.bar import bar as tricksy2

def nop(foo):
    # type: (MyClass) -> bar
    return bar()
class MyClass: pass
```

Reply:
Actually... this may not actually be testing much... ignore it for now. I think I actually need to understand the link between the mypy data is fed in, and how `find_import_info` actually gets called...